sycl/source/detail/scheduler/commands.cpp

```diff
@@ -3268,7 +3268,8 @@ ur_result_t ExecCGCommand::enqueueImpQueue() {
       // Kernel only uses assert if it's non interop one
       bool KernelUsesAssert =
```

Comment:
Why is checking for assert a run-time check?  Shouldn't we know at compile-time whether a kernel uses assert?

Reply:
This is because this information is extracted from the binary image. In theory, we could probably have the compiler communicate that to us via an integration header/built-in, but I expect the difference between that and caching it in this manner to be minimal.

Reply:
Ok.  I would personally prefer that we do as much at compile-time as possible, but I don't want to hold up this PR waiting for new built-ins to be implemented.  We can revisit this if it is still a problem later on.